File: sorting/problem268/problem268.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n = static_cast<int>(nums.size());
        int x = n;
        for (int i = 0; i < n; ++i) {
            x ^= i ^ nums[i];
        }
        return x;
    }
};

File: sorting/problem268/problem268.py
from typing import List


class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        # XOR indices 0..n-1 and values with n: pairs (i, nums[i]) cancel except missing
        n = len(nums)
        x = n
        for i, v in enumerate(nums):
            x ^= i ^ v
        return x
